kivy/core/window/__init__.py

```diff
@@ -2029,6 +2030,18 @@ def on_joy_button_up(self, stickid, buttonid):
         .. versionadded:: 1.9.0'''
         pass
 
+    def on_joy_added(self, deviceid):
+        '''Event called when a joystick has been plugged in.
+
+        .. versionadded:: 3.0.0'''
+        pass
+
+    def on_joy_removed(self, deviceid):
```

Comment:
why joy instead of "gamepad" ? for compatibility, with old descriptors - if someone is using it, we wont break it